### PR TITLE
Support custom mime map file

### PR DESCRIPTION
**Pull request by nakulkar-msft**

### CODE REVIEW

File: common/environment.go [hunk 1]
```diff
@@ -352,3 +353,11 @@ func (EnvironmentVariable) DisableSyslog() EnvironmentVariable {
 			"However, to reduce the noise in Syslog/Windows Event Log, consider setting this environment variable to true.",
 	}
 }
+
+func (EnvironmentVariable) MimeMapping() EnvironmentVariable {
+	return EnvironmentVariable {
+		Name:     "AZCOPY_CONTENT_TYPE_MAP",
+		DefaultValue: "",
+		Description: "Location of the file to override default OS mime mapping",
```
Comment (zezha-msft) [inline common/environment.go:361]: Give example of file format.

**Review by zezha-msft — APPROVED** [review]
lgtm, please add some info on the file format.

File: ste/mgr-JobPartMgr.go [hunk 2]
```diff
@@ -640,6 +642,9 @@ var builtinTypes = map[string]string{
 func (jpm *jobPartMgr) inferContentType(fullFilePath string, dataFileToXfer []byte) string {
 	fileExtension := filepath.Ext(fullFilePath)
 
+	if contentType, ok := environmentMimeMap[strings.ToLower(fileExtension)]; ok {
+		return contentType
+	}
```
Comment (mohsha-msft) [inline ste/mgr-JobPartMgr.go:647]: Ohh I see. 
1. Try to infer MIMEType from user specified file. (which we've added now)
2. Then try to infer from built-In-Types
3. Then try to infer from the mime.go lib provided by golang
4. And lastly from the first 512B (headers) of the file.

**Review by mohsha-msft — APPROVED** [review]
Looks pretty straight forward.

File: common/environment.go (hunk 1 above)
Comment (nakulkar-msft) [reply common/environment.go:361]: I'm adding a wiki.

**Pull request merged**